feat: add useBlockExplorerNavigation hook and integrate into EmptyHistory and TxHistoryListView components OK-42292

yikZero opened the pull request:
- Implemented useBlockExplorerNavigation hook to determine internal navigation status based on network and wallet ID.
- Updated EmptyHistory component to utilize the new hook for conditional rendering of button icon.
- Modified TxHistoryListView component to incorporate the hook for consistent button behavior across transaction history views.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Unified address navigation: tapping an address opens the block explorer directly, or—if a network must be chosen first—prompts for network selection before proceeding.
- Style
  - Action buttons updated to medium size with increased spacing for improved readability.
  - Context-aware icon: external link icon shown only when directly opening the block explorer (hidden when network selection is required).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yikZero committed:
feat: add useBlockExplorerNavigation hook and integrate into EmptyHistory and TxHistoryListView components OK-42292

- Implemented useBlockExplorerNavigation hook to determine internal navigation status based on network and wallet ID.
- Updated EmptyHistory component to utilize the new hook for conditional rendering of button icon.
- Modified TxHistoryListView component to incorporate the hook for consistent button behavior across transaction history views.

yikZero committed:
refactor: simplify conditional logic in EmptyHistory and TxHistoryListView components

- Updated handleOnPress functions in both EmptyHistory and TxHistoryListView components to use isInternalNav for conditional rendering, removing unnecessary checks for network and wallet ID.
- Improved code readability and maintainability by streamlining dependencies in useCallback hooks.

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/ee43a323-4164-4240-8061-1a6a430cbf30?refs=f272219cc8d8df546c519a30172984022ac73826&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/ee43a323-4164-4240-8061-1a6a430cbf30/license?refs=f272219cc8d8df546c519a30172984022ac73826&source=prChecksComment)

originalix commented on packages/kit/src/components/Empty/EmptyHistory.tsx:
```diff
@@ -64,15 +63,15 @@ function EmptyHistory({
-    network?.isAllNetworks,
-    network?.id,
-    walletId,
+    isInternalNav,
```
应该把下面的跳转行为放入 hooks，导出 openExplorer 的函数， 而不是只导出 `isInternalNav` 的判断，并且这个变量名语义化不好

```
appNavigation.pushModal(EModalRoutes.WalletAddress, {
        screen: EModalWalletAddressRoutes.WalletAddress,
        params: {
          title: intl.formatMessage({
            id: ETranslations.global_select_network,
          }),
          accountId,
          walletId: walletId ?? '',
          indexedAccountId: indexedAccountId ?? '',
          actionType: EWalletAddressActionType.ViewInExplorer,
        },
      });
    } else {
      await openExplorerAddressUrl({
        networkId: account?.createAtNetwork ?? network?.id,
        address: account?.address,
      });
    }
    ```

yikZero committed:
refactor: enhance useBlockExplorerNavigation hook and streamline EmptyHistory and TxHistoryListView components

- Refactored useBlockExplorerNavigation hook to include openExplorer function for handling navigation logic based on network selection.
- Updated EmptyHistory and TxHistoryListView components to utilize the new openExplorer function, simplifying the handleOnPress logic and improving code clarity.
- Removed unnecessary imports and dependencies, enhancing overall maintainability.

yikZero committed:
Merge branch 'x' into history-ui
